### PR TITLE
eigen: fetch source from gitlab

### PR DESCRIPTION
**Pull request by eadwu**
@svanderburg @7c6f434c
###### Motivation for this change
Migrated over to Gitlab.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by eadwu** [comment]
There seems to be a lot of rebuilds just from changing the `src`. I guess this would be better after a new version releases?

File: pkgs/development/libraries/eigen/2.0.nix [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
+    sha256 = "0d4knrcz04pxmxaqs5r3wv092950kl1z9wsw87vdzi9kgvc6wl0b";
   };
 
   nativeBuildInputs = [ cmake ];
 
   meta = with stdenv.lib; {
     description = "C++ template library for linear algebra: vectors, matrices, and related algorithms";
     license = licenses.lgpl3Plus;
-    homepage = http://eigen.tuxfamily.org ;
+    homepage = https://eigen.tuxfamily.org;
```
Comment (markuskowa) [inline pkgs/development/libraries/eigen/2.0.nix]:
```suggestion
    homepage = "https://eigen.tuxfamily.org";
```

File: pkgs/development/libraries/eigen/default.nix [hunk 2]
```diff
@@ -22,7 +20,7 @@ stdenv.mkDerivation {
   meta = with stdenv.lib; {
     description = "C++ template library for linear algebra: vectors, matrices, and related algorithms";
     license = licenses.lgpl3Plus;
-    homepage = http://eigen.tuxfamily.org ;
+    homepage = https://eigen.tuxfamily.org;
```
Comment (markuskowa) [inline pkgs/development/libraries/eigen/default.nix]:
```suggestion
    homepage = "https://eigen.tuxfamily.org";
```

**Review by veprbl — APPROVED** [review]
Diff between old unpacked tarball and the new source is minimal
```
# diff -aru eigen-eigen-b23437e61a07 /nix/store/6kix5k88g20hxhszdxh0kc4jpiz9smvb-source
Only in eigen-eigen-b23437e61a07: .hg_archival.txt
Only in eigen-eigen-b23437e61a07: .hgtags
```

**Pull request merged**